charts/opentelemetry-kube-stack/templates/collector.yaml

```diff
@@ -104,14 +104,18 @@ spec:
   observability:
   {{- toYaml . | nindent 4}}
   {{- end }}
-  {{- with $collector.updateStrategy }}
-  updateStrategy:
+  {{- if eq $collector.mode "daemonset" }}
+  {{- with (or $collector.daemonSetUpdateStrategy $collector.updateStrategy) }}
```

Comment:
Instead of setting daemonSetUpdateStrategy, `$collector.updateStrategy` should continue to set `updateStrategy` when set. In `NOTES.txt` we can add validation that enforces `updateStrategy` cannot be set when `deploymentUpdateStrategy` or  `daemonSetUpdateStrategy` are set.

Reply:
Unless in the operate `daemonSetUpdateStrategy` and `updateStrategy` are treated exactly the same.